pkg/convert/ptr.go

```diff
@@ -32,3 +32,16 @@ func SafeToUTC(t *time.Time) *time.Time {
 		return dt.In(time.UTC)
 	})
 }
+
+// Header represents generic primitives with a header like map, slice, array...
+type Header[E any] interface {
+	map[string]E | []E
+}
+
+// Safely dereference a pointer to a slice or map
+func DerefHeaderPtr[E any, T Header[E]](header *T) T {
```

Comment:
```suggestion
func DerefCompositeTypePtr[E any, T Header[E]](header *T) T {
```

The header is misleading for me. According to the [go spec](https://go.dev/ref/spec):

```
Composite types—array, struct, pointer, function, interface, slice, map, and channel types—may be constructed using type literals.
```

It's not a perfect match, but IMHO better then header.

Reply:
So what I was referring to was that they're nillable cause they point to their header struct in runtime
https://github.com/golang/go/blob/master/src/runtime/map.go#L109
https://github.com/golang/go/blob/master/src/runtime/slice.go (slice header has since been "removed", e.g. [see this depracation](https://pkg.go.dev/reflect#SliceHeader), now it's just slice)